src/coreclr/nativeaot/Runtime/windows/PalRedhawkMinWin.cpp

```diff
@@ -53,13 +53,45 @@ static uint32_t g_flsIndex = FLS_OUT_OF_INDEXES;
 void __stdcall FiberDetachCallback(void* lpFlsData)
 {
     ASSERT(g_flsIndex != FLS_OUT_OF_INDEXES);
+    ASSERT(g_flsIndex != NULL);
     ASSERT(lpFlsData == FlsGetValue(g_flsIndex));
 
-    if (lpFlsData != NULL)
+    // The current fiber is the home fiber of a thread, so the thread is shutting down
+    RuntimeThreadShutdown(lpFlsData);
+}
+
+REDHAWK_PALEXPORT bool REDHAWK_PALAPI PalInitComAndFlsSlot()
+{
+    ASSERT(g_flsIndex == FLS_OUT_OF_INDEXES);
+
+    // Making finalizer thread MTA early ensures that COM is initialized before we initialize our thread
+    // termination callback.
+    CoInitializeEx(NULL, COINIT_MULTITHREADED);
```

Comment:
We can check for bad HRESULT from CoInitializeEx as well.

Reply:
Can COM be somehow unavailable to the app? Like turned off for the platform or the app? 

I think in such case we'd want to ignore a failure and continue, since we do not actually require that init succeeded, just want to be sure it succeeds early enough.

Reply:
COM is part of Windows. Windows without COM is not a thing.

The current COM init logic propagates the errors from CoInitialize: 
https://github.com/dotnet/runtime/blob/88494801cdf4ddb496e78bf4d9165fe7f9273206/src/coreclr/nativeaot/System.Private.CoreLib/src/System/Threading/Thread.NativeAot.Windows.cs#L320-L335